pages/api/auth/signin/authorize.ts

```diff
@@ -0,0 +1,92 @@
+import { Admin, Doctor, Role, User } from "@prisma/client";
+import hasher from "../../../../utils/hasher/BcryptjsHasher";
+import { prisma } from "../../../../lib/prisma";
+import { InvalidCredentials } from "./invalidCredentials";
+
+type TSignInCredentials = {
+  email: string;
+  password: string;
+};
+
+function validate(credentials: TSignInCredentials): TSignInCredentials {
```

Comment:
I think this function is unecessary. NextAuth credentials example is a lot simple, and that has some reasons: 
1) if you receive a false/null/strage user it will not be retreived from the database (so, no need to validate)
2) if you receive an invalid password it will fail in hasher.compareAsync (so, no need to validate)